caddyhttp/httpserver/server.go

```diff
@@ -380,6 +380,10 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	status, _ := s.serveHTTP(w, r)
 
+	if status >= 200 {
```

Comment:
What if the status is `3xx` or even `5xx`? Feels like it could cause unintended side-effects, but maybe I'm misunderstanding.
```suggestion
	if status >= 200 && status < 300 {
```
Thoughts @mholt?